Automattic/wp-calypso
# Site Settings: refactor and update support links

## Changes proposed in this Pull Request
This started out as a simple PR to update the links to the support guide links and during the process, I decided to refactor a few things and make it a little cleaner and more uniform. Here are the major changes I made:
<img width="1075" alt="Markup 2022-02-16 at 20 25 07" src="https://user-images.githubusercontent.com/33258733/154340516-ff3062ed-11e7-4778-8827-6af059a65649.png">

#### Main Index
* Moved section headings within the appropriate components
* Removed unused props
* Added `siteIsAutomatedTransfer` prop to the sections that needed to provide the different support links based on that context.

#### Composing section
* Removed unnecessary connect to the store
* Removed unused props
* Added section header
* Changed `Latex` and `Shortcodes` components from class components to functions
* Updated support links
<img width="1629" alt="Markup 2022-02-16 at 20 30 35" src="https://user-images.githubusercontent.com/33258733/154341299-b10edddf-9b27-4075-ab37-45ecc50f86b1.png">

#### Content Types
* Extrapolated each fieldset into a component `BlogPosts`, `Testimonials`, and `Portfolios`.
* I tried to convert the index from a class to a function but was not successful with the autosave/update feature so I left this as a class.
* Updated support links
<img width="1626" alt="Markup 2022-02-16 at 20 31 05" src="https://user-images.githubusercontent.com/33258733/154341550-329fdc1f-a392-4ab0-8dad-636149a3ced1.png">

#### Masterbar (only in self-hosted Jetpack sites)
* Added section header

#### Media (only in self-hosted Jetpack sites)
* Added section header

#### Publishing Tools
* Updated support links
* Replaced the generic HTML elements with the standardized Form components to make things consistent
* Adjusted section header
<img width="1637" alt="Markup 2022-02-16 at 20 32 00" src="https://user-images.githubusercontent.com/33258733/154341592-bac4933f-8969-4c2c-be66-c217aa246cd9.png">

#### Theme Enhancements
* Changed from class to function component
* Removed unused props
* Updated support links
<img width="1636" alt="Markup 2022-02-16 at 20 29 05" src="https://user-images.githubusercontent.com/33258733/154341085-912db1c4-e187-4c9d-8e74-f0af75fa7f62.png">

#### Widgets
* Removed unused props
* Changed from class to function component

## Testing instructions
* Pull branch and launch calypso with `yarn start`
* Go to Settings ⇢ Writing
* Adjust all settings and options and verify they update correctly
* Visually inspect for errors (there are slight variations due to the update to match styling)
* Click the `( i )` and make sure the link goes to Jetpack links on JPOP only sites and WPCOM for Simple and AT sites
* Repeat this process with each version - Simple, AT, and JPOP site
* I tested side-by-side with the live version. After a change in my test environment, I would refresh my live version to make sure the changes applied to the site settings.

Fixes #59977
